lib/Doctrine/ORM/Mapping/JoinTableMapping.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+
+use function array_map;
+use function in_array;
+
+/** @template-implements ArrayAccess<string, mixed> */
+final class JoinTableMapping implements ArrayAccess
+{
+    use ArrayAccessImplementation;
+
+    public bool|null $quoted = null;
+
+    /** @var list<JoinColumnData> */
+    public array|null $joinColumns = null;
```

Comment:
The native type allows `null` while the `@var` type doesn't. This should be fixed. Are `null` and `[]` actually different cases? If not, I suggest to remove the nullability here.

Reply:
`[]` causes an exception from DBAL on this piece of code:

```
        if (count($foreignKey->getLocalColumns()) === 0) {
            throw new InvalidArgumentException("Incomplete definition. 'local' required.");
        }
```

So I guess they are different cases, and `[]` is not supposed to ever happen.

Reply:
Disallowing `null` seems OK according to the test suite, defaulting to `[]` isn't, though… so I guess it's not `null`, but uninitialized? I'm confused now…

Reply:
> so I guess it's not `null`, but uninitialized? I'm confused now…

What we're looking for is a representation for an incomplete `JoinTableMapping`. You construct that object and then you add columns to it. For that matter, the empty array seems like a good initial value. `null` is just an unnecessary edge case that static analysis will always force us to cover.

Reply:
I took a closer look at the failing test, it fails because of what's in `ManyToManyOwningSideMapping`:

```
        if (! isset($mapping->joinTable->joinColumns)) {
            $mapping->joinTable->joinColumns = [
                JoinColumnMapping::fromMappingArray([
                    'name' => $namingStrategy->joinKeyColumnName($mapping->sourceEntity, $selfReferencingEntityWithoutJoinColumns ? 'source' : null),
                    'referencedColumnName' => $namingStrategy->referenceColumnName(),
                    'onDelete' => 'CASCADE',
                ]),
            ];
        }
```

Using `=== []` instead of `isset()` fixes this particular issue, I'm looking at the next one.

Reply:
fixed :)